ui/apps/platform/src/services/AdministrationEventsService.ts

```diff
@@ -35,20 +40,34 @@ export type AdministrationEventType =
     | 'ADMINISTRATION_EVENT_TYPE_GENERIC'
     | 'ADMINISTRATION_EVENT_TYPE_LOG_MESSAGE';
 
+const types: AdministrationEventType[] = [
+    'ADMINISTRATION_EVENT_TYPE_UNKNOWN',
+    'ADMINISTRATION_EVENT_TYPE_GENERIC',
+    'ADMINISTRATION_EVENT_TYPE_LOG_MESSAGE',
+]; // for isType function
+
 export type AdministrationEventLevel =
     | 'ADMINISTRATION_EVENT_LEVEL_UNKNOWN'
     | 'ADMINISTRATION_EVENT_LEVEL_INFO'
     | 'ADMINISTRATION_EVENT_LEVEL_SUCCESS'
     | 'ADMINISTRATION_EVENT_LEVEL_WARNING'
     | 'ADMINISTRATION_EVENT_LEVEL_ERROR';
```

Comment:
If you are worried about this and `levels` getting out of sync:
```ts
export type AdministrationEventLevel = (typeof levels)[number];
```

I noticed we tend to repeatedly use a pattern where we create:
1. A `const` array of strings
2. A union type of all of those strings
3. A type predicate to narrow the type of a plain string to the above defined type

I haven't had a chance to think about it deeply, but I wonder if there is a clean pattern to avoid the boilerplate in creating these. 🤔

Reply:
Yes, you are intuitive. I was worried.

Is this an example which includes `as const` idiom for the array assignment?